[ONNX] replace AT_ASSERT with TORCH_INTERTNAL_ASSERT take 2

titaiwangms opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #86405

Address the AT_ASSERT in torch/jit/csrc/serialization (ONNX related).

titaiwangms committed:
[ONNX] replace AT_ASSERT with TORCH_INTERTNAL_ASSERT take 2

[ghstack-poisoned]

titaiwangms committed:
Update on "[ONNX] replace AT_ASSERT with TORCH_INTERTNAL_ASSERT take 2"


Address the AT_ASSERT in torch/jit/csrc/serialization (ONNX related).

[ghstack-poisoned]

titaiwangms committed:
Update on "[ONNX] replace AT_ASSERT with TORCH_INTERTNAL_ASSERT take 2"


Address the AT_ASSERT in torch/jit/csrc/serialization (ONNX related).

[ghstack-poisoned]

titaiwangms committed:
Update on "[ONNX] replace AT_ASSERT with TORCH_INTERTNAL_ASSERT take 2"


Address the AT_ASSERT in torch/jit/csrc/serialization (ONNX related).

[ghstack-poisoned]

justinchuby reviewed (approved):
Thanks!

BowenBao commented on torch/csrc/jit/passes/onnx/pattern_conversion/pattern_conversion.cpp:
```diff
@@ -146,7 +146,7 @@ std::unordered_map<int64_t, ConvertedIndex> MergeSliceAndSelectToIndices(
       dim_offset++;
     } else {
-      AT_ERROR(
+      TORCH_INTERNAL_ASSERT(
```
```suggestion
      TORCH_CHECK(false,
```

Use `TORCH_CHECK` for user input error. Also note the first argument of both `TORCH_CHECK` and `TORCH_INTERNAL_ASSERT` is for condition. `AT_ERROR` on the other hand only takes message string as argument.

AllenTiTaiWang replied:
Done

BowenBao reviewed (approved):
🚀  with inline comment.

titaiwangms committed:
Update on "[ONNX] replace AT_ASSERT with TORCH_INTERTNAL_ASSERT take 2"


Address the AT_ASSERT in torch/jit/csrc/serialization (ONNX related).

[ghstack-poisoned]

AllenTiTaiWang commented:
@pytorchbot merge -g
